starlette/requests.py

```diff
@@ -250,8 +251,13 @@ async def form(self) -> FormData:
             content_type_header = self.headers.get("Content-Type")
             content_type, options = parse_options_header(content_type_header)
             if content_type == b"multipart/form-data":
-                multipart_parser = MultiPartParser(self.headers, self.stream())
-                self._form = await multipart_parser.parse()
+                try:
+                    multipart_parser = MultiPartParser(self.headers, self.stream())
+                    self._form = await multipart_parser.parse()
+                except MultiPartException as exc:
+                    if "app" in self.scope:
+                        raise HTTPException(status_code=400, detail=exc.message)
```

Comment:
I don't want to use `exc.args[0]` here. Do you have a better solution? @adriangb

Reply:
I can have two different exceptions as well, and don't use `exc.message`...

Reply:
Ah I see where this was going now! Sorry for not catching on. Seeing this I think that `message` is okay, but 2 exceptions are also a good idea, I think there's precedent for both. I'll leave it up to you.